client-js/src/client/backend-client.js

```diff
@@ -28,83 +116,103 @@
  */
 export class BackendClient {
 
-    /**
-     * Creates a new `BackendClient`.
-     *
-     * @param httpClient          the {@link HttpClient} to connect to
-     *                            the backend with
-     * @param firebaseClient      the {@link FirebaseClient} to read the query
-     *                            results with
-     * @param actorRequestFactory the {@link ActorRequestFactory} to instantiate
-     *                            the actor requests with
-     */
-    constructor(httpClient, firebaseClient, actorRequestFactory) {
-        this._httpClient = httpClient;
-        this._firebase = firebaseClient;
-        this._actorRequestFactory = actorRequestFactory;
-    }
-
-    /**
-     * Fetches all the entities of the given type.
-     *
-     * @param type          the target {@link TypeUrl}
-     * @param dataCallback  the callback which receives the data, one-by-one, in
-     *                      a form of a JS object
-     * @param errorCallback the callback which receives the errors
-     */
-    fetchAll(type, dataCallback, errorCallback = null) {
-        let query = this._actorRequestFactory.queryAll(type);
-        this._fetch(query, dataCallback, errorCallback);
-    }
+  /**
+   * @param {!HttpEndpoint} endpoint the server endpoint to execute queries and commands
+   * @param {!FirebaseClient} firebaseClient the client to read the query results from
+   * @param {!ActorRequestFactory} actorRequestFactory a factory to instantiate the actor requests with
+   */
+  constructor(endpoint, firebaseClient, actorRequestFactory) {
+    this._endpoint = endpoint;
+    this._firebase = firebaseClient;
+    this._actorRequestFactory = actorRequestFactory;
+  }
 
-    /**
-     * Fetches a single entity of the given type.
-     *
-     * @param type          the target {@link TypeUrl}
-     * @param id            the target entity ID, as a {@link TypedMessage}
-     * @param dataCallback  the callback which receives the single data item, in
-     *                      a form of a JS object
-     * @param errorCallback the callback which receives the errors
-     */
-    fetchById(type, id, dataCallback, errorCallback = null) {
-        let query = this._actorRequestFactory.queryById(type.value, id);
-        this._fetch(query, dataCallback, errorCallback);
-    }
+  /**
+   * Defines a fetch query of all entities matching the filters provided as arguments.
+   * This fetch is executed later upon calling 
+   *
+   * @param {!TypeUrl} ofType a type of the entities to be queried
+   * @returns {BackendClient.Fetch} a fetch object allowing to specify additional remote
+   *                                call parameters and executed the query.
```

Comment:
> additional remote call parameters...

Can the invoker actually do something on the returned object other than running the query with one of the available strategies?